src/ui/Facet/Facet.ts

```diff
@@ -1251,7 +1251,7 @@ export class Facet extends Component {
 
   protected updateAppearanceDependingOnState() {
     $$(this.element).toggleClass('coveo-active', this.values.hasSelectedOrExcludedValues());
-    if (!$$(this.element).hasClass('coveo-with-placeholder')) {
+    if (!$$(this.element).hasClass('coveo-with-placeholder') || this.nbAvailableValues < 1) {
```

Comment:
I would remove the check in javascript for `coveo-with-placeholder`, and simply ensure that, in css : 

`coveo-facet-empty` is hidden.
`coveo-with-placeholder` is visible.
`coveo-facet-empty coveo-with-placeholder` is visible